src/rdkafka_cgrp.c

```diff
@@ -2987,6 +3624,57 @@ rd_kafka_cgrp_assign (rd_kafka_cgrp_t *rkcg,
 }
 
 
+/**
+ * @brief Construct a typed map from list \p list with key corresponding to
+ *        each element in the list and value NULL.
+ */
+static map_toppar_member_info_t *
+rd_kafka_toppar_list_to_map (rd_kafka_topic_partition_list_t *list) {
+        int i;
+
+        if (!list)
+                return NULL;
```

Comment:
it does makes usage cleaner, because there is a case when the input is NULL, but i'll move the logic out to the caller as you wish. i do agree that makes better sense for a general purpose function.

a bigger issue here is C's type system isn't good enough to let you know what's going on.